### ISSUE TITLE
The issue regarding backpropagation in the coarse matching module and the fine matching module

### ISSUE THREAD

**Issue by DiopWang** [leoluxxx/JamMa#38]
Dear author:
Thanks for your great job. I am confused about the part of backpropagation, Could you please explain whether the parameters of the fine-matching module are updated in reverse and propagated back to the parameters of the coarse-matching module after the final matching is completed? Or, is this fine-grained matching module directly reversed and fed back into ConvnextV2?

**Comment by leoluxxx** [comment]
Hello, the gradient of fine matching will be backpropagated to coarse features because coarse and fine features are first fused before fine matching is performed.

**Comment by DiopWang** [comment]
> Hello, the gradient of fine matching will be backpropagated to coarse features because coarse and fine features are first fused before fine matching is performed.

Hello, thanks for your reply! But what I mean is that after calculating the loss for fine-grained matching and the sub_loss, will it return to update the parameters of the coarse matching module? And I see this
https://github.com/leoluxxx/JamMa/blob/24acce206cdd09f60465459d7112c83f09c75f15/src/jamma/matching_module.py#L113

**Comment by leoluxxx** [comment]
The @torch.no_grad() part simply generates coarse matches based on the conf_matrix and does not involve any learnable parameters. 
After feature windows are cropped from coarse matches, the gradients of fine matching loss and sub loss are propagated through the cropped feature windows (fusion of coarse and fine features).
